pkgs/desktops/gnome-3/extensions/update-extensions.py

```diff
@@ -0,0 +1,136 @@
+        if not extension_version:
+            continue
+        print(f"[{shell_version}] Downloading {uuid}, {str(extension_version)}")
+        sha256 = fetch_sha256sum(uuid, str(extension_version))
+        extension_versions[shell_version] = {
+            "version": str(extension_version),
+            "sha256": sha256,
+        }
+    return extension_versions
+
+
+# Download the extension and hash it
+def fetch_sha256sum(uuid: str, version: str) -> str:
+    import hashlib, base64
+
+    uuid = uuid.replace("@", "")
+    url: str = f"https://extensions.gnome.org/extension-data/{uuid}.v{version}.shell-extension.zip"
+    remote = urllib.request.urlopen(url)
+    hasher = hashlib.sha256()
+
+    while True:
+        data = remote.read(4096)
+        if not data:
+            break
+        hasher.update(data)
+    return hasher.hexdigest()
+
+
+# Fetching extensions
+page = 0
+extensions = []
+while True:
+    page += 1
+    print("Scraping page " + str(page))
+    try:
+        with urllib.request.urlopen(
+            f"https://extensions.gnome.org/extension-query/?n_per_page=25&page={page}"
+        ) as url:
+            data = json.loads(url.read().decode())["extensions"]
+            responseLength = len(data)
+            print(f"Found {responseLength} extensions")
+
+            for extension in data:
+                # Yeah, there are some extensions without any releases
+                if not extension["shell_version_map"]:
+                    continue
+
+                print("Processing " + extension["uuid"])
+
+                # Throw away that 'pk' key. What is it for?
+                extension["shell_version_map"] = {
+                    k: v["version"] for k, v in extension["shell_version_map"].items()
+                }
+                # Transform shell_version_map to be more useful for us. Also throw away unwanted versions
+                extension["shell_version_map"] = generate_extension_versions(
+                    extension["shell_version_map"], extension["uuid"]
+                )
+                if not extension["shell_version_map"]:
+                    # No compatible versions found
+                    continue
+
+                # Parse something like "/extension/1475/battery-time/" and output "battery-time-1475"
+                def pname_from_url(url: str) -> str:
+                    url = url.split("/")  # type: ignore
+                    return url[3] + "-" + url[2]
+
+                extension["pname"] = pname_from_url(extension["link"])
+                extension["link"] = "https://extensions.gnome.org" + extension["link"]
+
+                # Replace \u0123 in strings. This is required for nix < 2.3.10
+                # extension["name"] = extension["name"].encode('unicode-escape').decode()
+                # extension["description"] = extension["description"].encode('unicode-escape').decode()
+
+                # Remove unused keys
+                del extension["screenshot"]
```

Comment:
Maybe it would be easier to delete all the keys but a few that we explicitly list? I imagine this set might grow/change over time. We shouldn't track what we do not care about.